GauchoGrub/app/src/main/java/edu/ucsb/cs/cs190i/gauchogrub/gauchogrub/utils/BootReceiver.java

```diff
@@ -29,32 +32,62 @@ public BootReceiver() {
      */
     @Override
     public void onReceive(Context context, Intent intent) {
-        AlarmManager alarmManager = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
-        ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
-        NetworkInfo netinfo = cm.getActiveNetworkInfo();
-        // Prepares intent for data automation service
-        Intent timedIntent = new Intent(context, MenuScraperService.class);
-        PendingIntent pendingAutomationIntent = PendingIntent.getService(context, 0, timedIntent, 0);
+        if (intent.getAction().equals("android.intent.action.BOOT_COMPLETED")) {
+            AlarmManager alarmManager = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+
+            // sets up notification service
+            startNotificationService(context, alarmManager);
+
+            // sets up data automation service
+            startMenuScraperService(context, alarmManager);
+        }
+    }
 
-        // sets up notification service
+    private void startNotificationService(Context context, AlarmManager alarmManager) {
         Intent notificationIntent = new Intent(context, NotificationService.class);
         PendingIntent pendingNotificationIntent = PendingIntent.getService(context, 0, notificationIntent, 0);
-        Calendar calendar = Calendar.getInstance();
-        calendar.set(Calendar.HOUR_OF_DAY, 8);
-        calendar.set(Calendar.MINUTE, 0);
-        calendar.set(Calendar.SECOND, 0);
-        alarmManager.setRepeating(AlarmManager.RTC, calendar.getTimeInMillis(),
+        Calendar notificationCalendar = Calendar.getInstance();
```

Comment:
Here, Calendar can be replaced with LocalDateTime:
`LocalDateTime.now().withHourOfDay(7).withMinuteOfHour(0).withSecondOfMinute(0);`